src/Columns/MaskOperations.cpp

```diff
@@ -120,6 +123,10 @@ size_t extractMaskNumericImpl(
 
         mask[i] = value;
     }
+
+    if (i != mask_size || data_index != data_size)
```

Comment:
Probably need another check where `column_is_short` is false.

Reply:
Yep, forgot to check `column_is_short`. Updated